### DISCUSSION TITLE
Auto incline on Bowflex 25 treadmill

### DISCUSSION THREAD

**Discussion by luvictu** [cagnulein/qdomyos-zwift#2451] [Q&A]
hi Roberto, hope all is well.

I've managed to connect my Bowflex 25 treadmill to Zwift (on Android tablet) via qdomyos (on Android phone), it all works fine but auto-incline does not seem yp work. I've tried the Bike/MeetUp workaround, the incline icon on QZ flashes while in bike model but stops as soon as I jump into the run meetup).
It seems that the treadmill JRNY platform does not allow auto incline actually, so not sure if this is a dead-end, or if somebody has succeed with this same treadmill model? Been searching on the forums for a couple of days but cannot figure this out.

 Any help would be really appreciated. Thank you!

**Comment by cagnulein** [comment]
hi! did you set your zwift credentials on qz?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 15 lug 2024 alle 12:47 luvictu ***@***.***> ha
scritto:

**Comment by luvictu** [comment]
hey thanks for the speedy reply Roberto. Yes I did set the Zwift options on qz, including my zwift credentials. I'm assuming here that this should be working? :-)

**Reply by cagnulein** [reply]
second question: are you able to change inclination manually from QZ? Because  I guess this is the main reason. I don't know any Bowflex treadmill with auto incline possibility. In case you can add some relay to it and with a special version of QZ you can have auto inclination!

**Reply by luvictu** [reply]
No I can't change inclination from QZ, but I cannot change speed either, so not sure where the problem must be.
Don't want to hassle you too much with this, so I will keep trying things and searching on the facebook group etc to see if I can catch up with somebody else with Bowflex treadmills and see what I can find out. If there's no other workaround I would go for the relay route as you suggest and will ask for your help again :-) Thank you so much Roberto!

**Reply by cagnulein** [reply]
The limitation is on the firmware inside the treadmill. There is no workaround for this except commanding the physical buttons using 4 relays (speed up and down and inclination up and down). :)

**Reply by luvictu** [reply]
Oh I see, these are not good news really. Ok will keep going with manual incline until I figure out the relays. Thanks once more Roberto.

**Reply by luvictu** [reply]
Oh let me make the most of our conversation here, just out of curiosity: should we be able to have zwift and QZ running in the same Android device at the same time? Or this is only possible on PC?

**Reply by cagnulein** [reply]
if you want i have already the amazon links for these and the code is already ready. I did exactly for a bowflex users https://github.com/cagnulein/qdomyos-zwift/issues/2202

**Reply by cagnulein** [reply]
yes you can run qz and zwift on the same device. You have to enable the android notification icon in the experimental setttings in QZ to do so

**Reply by luvictu** [reply]
It's enabled already, but Zwift still cannot find any speed/power source (which I guess it should be this same phone right?)

**Comment by cagnulein** [comment]
you have to see a wahoo virtual device on zwift. don't you see it?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno lun 15 lug 2024 alle 13:59 luvictu ***@***.***> ha
scritto:

**Reply by luvictu** [reply]
No I don't, just a blank list and Zwift "searching..." :-)
So what I'm doing is: launch zwift, then launch Qz, go back to Zwift and look for a Power or Speed source either for Cycle or Run.... Doing this with Zwift on a separate android tablet works just fine, same on a PC.

**Reply by cagnulein** [reply]
do the opposite. Run qz, connect to the treadmill. when you see the tiles, open zwift. Also be sure you don't have any power saving settings enabled on your phone.
By the way which phone is it? it must have some gb of ram to run both at the same time

**Reply by luvictu** [reply]
Yes now it works, Zwift is finding a "Wahoo Tread 0003"  and a "wahoo kickr 0000 33", it connects with qz and the treadmill, so all good! Thanks!!

**Reply by cagnulein** [reply]
that's great! remember to rate the app on the store if you didn't already. It means a lot to me! Also join the swag bag from the app if you can!
Thanks!
